### PR TITLE
mapbox/driving-traffic support added to the list of directions profil…

### PR DESCRIPTION
**Pull request by srabenja**
…es, and checks added for maximum number of coordinates that can be provided, as discussed with @zugaldia

### CODE REVIEW

File: mapbox/libjava-services/src/main/java/com/mapbox/services/api/directions/v5/MapboxDirections.java [hunk 1]
```diff
@@ -464,6 +465,18 @@ public MapboxDirections build() throws ServicesException {
           "You should provide at least two coordinates (from/to).");
       }
 
+
+      if (profile != null && profile.equals(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
```
Comment (zugaldia) [inline mapbox/libjava-services/src/main/java/com/mapbox/services/api/directions/v5/MapboxDirections.java]: @srabenja Considering that `profile` and `coordinates` are required parameters, how about we add a check before these two raise an exception if either is `null` (or empty)?

**Comment by srabenja** [comment]
@zugaldia I added the changes you mentioned

**Comment by zugaldia** [comment]
@srabenja I see the fixture, but I see no tests using it?

**Comment by cammace** [comment]
I have fixed a few documentation things, cleaned up the fixture, rewrote some exception messages to make it more clear what error occurred and lastly, added a few test. @zugaldia can you review this again?

**Pull request merged**